core-bundle/src/Resources/contao/dca/tl_article.php

```diff
@@ -770,20 +770,17 @@ public function cutArticle($row, $href, $label, $title, $icon, $attributes)
 	 * @param array         $arrClipboard
 	 *
 	 * @return string
+	 *
+	 * @deprecated
 	 */
 	public function pasteArticle(DataContainer $dc, $row, $table, $cr, $arrClipboard=null)
 	{
-		$imagePasteAfter = Image::getHtml('pasteafter.svg', sprintf($GLOBALS['TL_LANG'][$dc->table]['pasteafter'][1], $row['id']));
-		$imagePasteInto = Image::getHtml('pasteinto.svg', sprintf($GLOBALS['TL_LANG'][$dc->table]['pasteinto'][1], $row['id']));
-
-		if ($table == $GLOBALS['TL_DCA'][$dc->table]['config']['ptable'])
-		{
-			return ($row['type'] == 'root' || !$this->User->isAllowed(BackendUser::CAN_EDIT_ARTICLE_HIERARCHY, $row) || $cr) ? Image::getHtml('pasteinto_.svg') . ' ' : '<a href="' . $this->addToUrl('act=' . $arrClipboard['mode'] . '&amp;mode=2&amp;pid=' . $row['id'] . (!is_array($arrClipboard['id']) ? '&amp;id=' . $arrClipboard['id'] : '')) . '" title="' . StringUtil::specialchars(sprintf($GLOBALS['TL_LANG'][$dc->table]['pasteinto'][1], $row['id'])) . '" onclick="Backend.getScrollOffset()">' . $imagePasteInto . '</a> ';
-		}
-
-		$objPage = PageModel::findById($row['pid']);
+		@trigger_error('Using "tl_article::pasteArticle()" has been deprecated and will no longer work in Contao 5.0. Use "ContentCompositionListener::renderArticlePasteButton()" instead.', E_USER_DEPRECATED);
```

Comment:
I disagree. This is an event listener. You're not supposed to use this ever.

Reply:
Changed in 1f70bdbeacefb095fa911bd2930bbcdd9ed709fd